### PR TITLE
[SHRINKWRAP-538] Resolving verbose or redundant code constructs

### PR DESCRIPTION
**Pull request by khermano**
Issue: https://issues.redhat.com/browse/SHRINKWRAP-538

### CODE REVIEW

File: api/src/main/java/org/jboss/shrinkwrap/api/Filters.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ private static Filter<ArchivePath> createRegExpFilter(String filterClassName, Pa
         StringBuilder classExpression = new StringBuilder();
         for (Package pack : packages) {
             classExpression.append("|");
-            classExpression.append("(.*" + pack.getName().replaceAll("\\.", "\\.") + ".*)");
+            classExpression.append("(.*").append(pack.getName().replaceAll("\\.", ".")).append(".*)");
```
Comment (petrberan) [inline api/src/main/java/org/jboss/shrinkwrap/api/Filters.java:156]: Why the change of `+` to `.append`? In other instances in this PR you're doing the opposite.

Wouldn't it be also better to have just a single line of `classExpression.append("|(.*" + pack.getName().replaceAll("\\.", ".")) + ".*)");`?

Comment (khermano) [reply api/src/main/java/org/jboss/shrinkwrap/api/Filters.java:156]: .append() is the recommended way how to work with StringBuilder. It avoids creating unnecessary intermediate String objects. On the other hand, the recommended way to work with String is to use "+".

Comment (petrberan) [reply api/src/main/java/org/jboss/shrinkwrap/api/Filters.java:156]: True, missed that. My comment regarding the single line append still stands though

Comment (khermano) [reply api/src/main/java/org/jboss/shrinkwrap/api/Filters.java:156]: That happens and I understand, but it's not a single line since it's inside the for loop.

File: api/src/main/java/org/jboss/shrinkwrap/api/Filters.java [hunk 2]
```diff
@@ -152,8 +152,7 @@ public static Filter<ArchivePath> include(Package... packages) {
     private static Filter<ArchivePath> createRegExpFilter(String filterClassName, Package... packages) {
         StringBuilder classExpression = new StringBuilder();
         for (Package pack : packages) {
-            classExpression.append("|");
-            classExpression.append("(.*" + pack.getName().replaceAll("\\.", "\\.") + ".*)");
+            classExpression.append("|").append("(.*").append(pack.getName().replaceAll("\\.", ".")).append(".*)");
```
Comment (petrberan) [inline api/src/main/java/org/jboss/shrinkwrap/api/Filters.java:155]: Still would very much prefer `.append("|(.*")` over `.append("|").append("(.*")`

**Review by petrberan — APPROVED** [review]
LGTM
